assets/app/features/upgrade/upgrade.controller.js

```diff
@@ -56,6 +70,8 @@
          */
         function cancelUpgrade() {
             upgradeFactory.cancelUpgrade().then(function(/* response */) {
+                // clear last page seen by the user
+                localStorage.removeItem(UPGRADE_LAST_STATE_KEY);
```

Comment:
The other case already used internals of `upgradeStepsFatory` which seemed out of place. Here we have something which is very specific to cancel feature and will likely change in the future. I would leave it like this for now.